File: sdks/ts/src/sdk/models/operations/postmetadatacascadepoliciesidapply.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostMetadataCascadePoliciesIdApplyPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=metadata_cascade_policy_id" })
  metadataCascadePolicyId: string;
}

export enum PostMetadataCascadePoliciesIdApplyRequestBodyConflictResolutionEnum {
    None = "none",
    Overwrite = "overwrite"
}


export class PostMetadataCascadePoliciesIdApplyRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=conflict_resolution" })
  conflictResolution: PostMetadataCascadePoliciesIdApplyRequestBodyConflictResolutionEnum;
}


export class PostMetadataCascadePoliciesIdApplyRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PostMetadataCascadePoliciesIdApplyPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostMetadataCascadePoliciesIdApplyRequestBody;
}


export class PostMetadataCascadePoliciesIdApplyResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: sdks/go/pkg/models/operations/postmetadatacascadepoliciesidapply.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PostMetadataCascadePoliciesIDApplyPathParams struct {
	MetadataCascadePolicyID string `pathParam:"style=simple,explode=false,name=metadata_cascade_policy_id"`
}

type PostMetadataCascadePoliciesIDApplyRequestBodyConflictResolutionEnum string

const (
	PostMetadataCascadePoliciesIDApplyRequestBodyConflictResolutionEnumNone      PostMetadataCascadePoliciesIDApplyRequestBodyConflictResolutionEnum = "none"
	PostMetadataCascadePoliciesIDApplyRequestBodyConflictResolutionEnumOverwrite PostMetadataCascadePoliciesIDApplyRequestBodyConflictResolutionEnum = "overwrite"
)

type PostMetadataCascadePoliciesIDApplyRequestBody struct {
	ConflictResolution PostMetadataCascadePoliciesIDApplyRequestBodyConflictResolutionEnum `json:"conflict_resolution"`
}

type PostMetadataCascadePoliciesIDApplyRequest struct {
	Retries    *utils.RetryConfig
	PathParams PostMetadataCascadePoliciesIDApplyPathParams
	Request    *PostMetadataCascadePoliciesIDApplyRequestBody `request:"mediaType=application/json"`
}

type PostMetadataCascadePoliciesIDApplyResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
}
